packages/amplify-cli/src/commands/post-install.ts

```diff
@@ -0,0 +1,45 @@
+import { $TSContext, GetPackageAssetPaths, pathManager } from 'amplify-cli-core';
+import { entries } from 'lodash';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+
+export const run = async (context: $TSContext) => {
+  // clean any previous libs
+  await fs.remove(pathManager.getAmplifyLibRoot());
+  await fs.ensureDir(pathManager.getAmplifyLibRoot());
+
+  const pluginMetaList = entries(context.pluginPlatform.plugins)
+    // can't use flatMap here as our target node version doesn't support it :(
+    .reduce(
+      (allPlugins, [_, categoryPlugins]) => allPlugins.concat(categoryPlugins),
+      [] as { packageName: string; packageLocation: string }[],
+    )
+    .concat(additionalPackageRegistry.map(packageName => ({ packageName, packageLocation: path.parse(require.resolve(packageName)).dir })));
+
+  await Promise.all(
```

Comment:
Wondering if this should be a serial operation rather than a parallel one to avoid filesystem overrides.

Reply:
what issues do you think would happen? The only issue I could see is if it tried to read / write to the same location at the same time but each plugin should only list files in it's package and files in each package are written to their own directory so there should never be any conflicts